### DISCUSSION TITLE
useMutation hook does not have the refetchOnReconnect?

### DISCUSSION THREAD

**Discussion by scc02** [TanStack/query#5205] [Q&A]
Hi
I'm trying to prevent useMutation refetch after re-gain network connection.
Any idea what I'm doing wrong?

Thanks!

**Comment by louis-young** [comment]
Hi 👋 

Do you want the mutation to run even when you're offline? I wonder if you can use the network mode to achieve this: https://tanstack.com/query/v4/docs/react/guides/network-mode

**Reply by scc02** [reply]
Yes, that's how I ended up setting it up. However, I'm curious to know if useMutation has an option like refetchOnReconnect in useQuery? 😂

**Reply by louis-young** [reply]
Awesome. No, it doesn't; there is no concept of refetching a mutation. You can see the useMutation API here: https://tanstack.com/query/v4/docs/react/reference/useMutation

**Reply by scc02** [reply]
Found this. https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-3#retryoffline-mutations
> If mutations `fail because the device is offline`, they will be `retried` in the same order `when the device reconnects`.

Howerver,It seems that there is no such option to control whether `retry`.

**Reply by louis-young** [reply]
That's an old migration guide. You can pass a function to retry to dictate the retry behaviour if that's what you're trying to do 🙂

**Reply by scc02** [reply]
Perhaps I didn't express my question clearly.I'm just wondering if it's possible for useMutation to not send a request after reconnecting (except configure networkMode)

**Reply by louis-young** [reply]
Do you want the mutation to run even when you're offline?

**Reply by scc02** [reply]
if execute `mutation.mutate` when offline, I want the mutation to not retry when the device reconnects. （do now want to config networkMode）
